server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -308,73 +311,68 @@ public String resolveDateMathExpression(String dateExpression) {
         return dateMathExpressionResolver.resolveExpression(dateExpression, new Context(null, null));
     }
 
+    /**
+     * Resolve an array of expressions to the set of indices and aliases that these expressions match.
+     */
+    public Set<String> resolveExpressions(ClusterState state, String... expressions) {
+        Context context = new Context(state, IndicesOptions.lenientExpandOpen(), true, false);
+        List<String> resolvedExpressions = Arrays.asList(expressions);
+        for (ExpressionResolver expressionResolver : expressionResolvers) {
+            resolvedExpressions = expressionResolver.resolve(context, resolvedExpressions);
```

Comment:
I think we should also fix the ExpressionResolver to take and return a Set<String>. I can see `WildcardExpressionResolver` is already converting from set to list internally. I am ok with doing this as a followup. Either way is fine

Reply:
Agreed, I added a TODO in WildcardExpressionResolver. I'd like to keep this change contained if possible since I intend to backport it to stable branches, so I'd rather like to do this as a follow-up.

Reply:
++